uniffi_bindgen/src/bindings/kotlin/templates/ObjectRuntime.kt

```diff
@@ -0,0 +1,161 @@
+// Interface implemented by anything that can contain an object reference.
+//
+// Such types expose a `destroy()` method that must be called to cleanly
+// dispose of the contained objects. Failure to call this method may result
+// in memory leaks.
+//
+// The easiest way to ensure this method is called is to use the `.use`
+// helper method to execute a block and destroy the object at the end.
+interface Disposable {
+    fun destroy()
+    companion object {
+        fun destroy(vararg args: Any?) {
+            args.filterIsInstance<Disposable>()
+                .forEach(Disposable::destroy)
+        }
```

Comment:
This method is new, and comes from the reason that when destructors are rendered we no longer have access to the `ComponentInterface`.

Some lifetime-fu was needed, or we could skip it and ask Kotlin to do the little bit of extra work.